pkg/controller/perconaservermongodb/vs.go

```diff
@@ -1,100 +1,87 @@
 package perconaservermongodb
 
 import (
-	"encoding/json"
 	"fmt"
 	"net/http"
 	"net/url"
 	"strings"
 	"time"
 
-	v1 "github.com/percona/percona-server-mongodb-operator/pkg/apis/psmdb/v1"
-)
+	"github.com/percona/percona-server-mongodb-operator/versionserviceclient/models"
 
-func (vs VersionServiceClient) GetExactVersion(endpoint string, vm VersionMeta) (DepVersion, error) {
-	client := http.Client{
-		Timeout: 5 * time.Second,
-	}
+	"github.com/percona/percona-server-mongodb-operator/versionserviceclient/version_service"
 
-	requestURL, err := url.Parse(
-		fmt.Sprintf("%s/v1/psmdb-operator/%s/%s",
-			strings.TrimRight(endpoint, "/"),
-			vs.OpVersion,
-			vm.Apply,
-		),
-	)
-	if err != nil {
-		return DepVersion{}, err
-	}
+	"github.com/percona/percona-server-mongodb-operator/versionserviceclient"
+)
 
-	q := requestURL.Query()
-	q.Add("databaseVersion", vm.MongoVersion)
-	q.Add("kubeVersion", vm.KubeVersion)
-	q.Add("platform", vm.Platform)
-	q.Add("customResourceUID", vm.CRUID)
+const productName = "psmdb-operator"
 
-	if vm.PMMVersion != "" {
-		q.Add("pmmVersion", vm.PMMVersion)
-	}
-
-	if vm.BackupVersion != "" {
-		q.Add("backupVersion", vm.BackupVersion)
+func (vs VersionServiceClient) GetExactVersion(endpoint string, vm VersionMeta) (DepVersion, error) {
+	if strings.Contains(endpoint, "https://check.percona.com/versions") {
+		endpoint = "https://check.percona.com"
 	}
 
-	requestURL.RawQuery = q.Encode()
-	req, err := http.NewRequest("GET", requestURL.String(), nil)
+	requestURL, err := url.Parse(endpoint)
 	if err != nil {
 		return DepVersion{}, err
 	}
 
-	req.Header.Set("Accept", "application/json")
-
-	resp, err := client.Do(req)
-	if err != nil {
-		return DepVersion{}, err
+	vsClient := versionserviceclient.NewHTTPClientWithConfig(nil, &versionserviceclient.TransportConfig{
+		Host:     requestURL.Host,
+		BasePath: requestURL.Path,
+		Schemes:  []string{requestURL.Scheme},
+	})
+
+	applyParams := &version_service.VersionServiceApplyParams{
+		Apply:             vm.Apply,
+		BackupVersion:     &vm.BackupVersion,
+		CustomResourceUID: &vm.CRUID,
+		DatabaseVersion:   &vm.MongoVersion,
+		KubeVersion:       &vm.KubeVersion,
+		OperatorVersion:   "1.5.0",
+		Platform:          &vm.Platform,
+		PmmVersion:        &vm.PMMVersion,
+		Product:           productName,
+		HTTPClient:        &http.Client{Timeout: 10 * time.Second},
 	}
+	applyParams = applyParams.WithTimeout(10 * time.Second)
 
-	defer resp.Body.Close()
-	if resp.StatusCode != http.StatusOK {
-		return DepVersion{}, fmt.Errorf("received bad status code %s", resp.Status)
-	}
+	resp, err := vsClient.VersionService.VersionServiceApply(applyParams)
```

Comment:
Should we check status code or in case of unsuccessful status code it will return error?

Reply:
Yea, if we fail to make http request, it will return error.
In case of resp - it's not an `http.Response`, it's custom response with decoded to struct data from server